### PR TITLE
Rework FFF/Hater

### PR DESCRIPTION
**Pull request by NikoCat233**
1.Add fff to reset cam list (WTF how it was done before)
2.move FFF codes to FFF.cs
3.Allow FFF to kill converted roles/convert roles to win (thewhiskas27's idea)
"https://discord.com/channels/1094344790910455908/1155823900765204562"
4.FFF can now misfire kill target

### CODE REVIEW

**Review by 0xDrMoe — APPROVED** [review]

**Comment by 0xDrMoe** [comment]
if you want, just check/fix this for us, @Gurge44

File: Patches/CheckGameEndPatch.cs [hunk 1]
```diff
@@ -274,16 +274,10 @@ public static bool Prefix()
 
 
                 //FFF
-                if (CustomWinnerHolder.WinnerTeam != CustomWinner.Lovers && !CustomWinnerHolder.AdditionalWinnerTeams.Contains(AdditionalWinners.Lovers) && !CustomRolesHelper.RoleExist(CustomRoles.Lovers) && !CustomRolesHelper.RoleExist(CustomRoles.Ntr))
+                if (FFF.winnerFFFList.Count > 0)
```
Comment (Gurge44) [inline Patches/CheckGameEndPatch.cs:277]: .Count > 0
can be replaced with
.Any()

File: Roles/Neutral/FFF.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        }
+
+        public static void Add(byte playerId)
+        {
+            playerIdList.Add(playerId);
+            IsEnable = true;
+
+            if (!AmongUsClient.Instance.AmHost) return;
+            if (!Main.ResetCamPlayerList.Contains(playerId))
+                Main.ResetCamPlayerList.Add(playerId);
```
Comment (Gurge44) [inline Roles/Neutral/FFF.cs:56]: You can remove this line as Hater doesn't keep the game going, and therefore it doesn't need to trigger anti blackout

Comment (Gurge44) [reply Roles/Neutral/FFF.cs:56]: I can PR these changes in like half an hour

**Pull request merged**